src/Conversion/ONNXToTorch/ONNXToTorchCommon.cpp

```diff
@@ -16,34 +16,80 @@
 //===-----------------------------------------------------------------===//
 
 TorchTypeConverter::TorchTypeConverter() {
-  // The order of type conversion is important: later ones are tried earlier.
+  /// The order of type conversion is important: later ones are tried earlier.
+  /// Legalize all remaining types to ensure successful conversions.
   addConversion([](Type type) { return type; });
 
-  addConversion([](StringType stringType) {
-    return Torch::StringType::get(stringType.getContext());
-  });
-
-   // Use UnrealizedConversionCast as the bridge so that we don't need to pull in
-   // patterns for other dialects.
-  addSourceMaterialization([&](OpBuilder &builder, Type resultType,
+  /// Unrealized conversion cast
+  auto addUnrealizedCast = [&](OpBuilder &builder, Type resultType,
                                ValueRange inputs,
                                Location loc) -> Optional<Value> {
     if (inputs.size() != 1)
       return llvm::None;
     return builder.create<UnrealizedConversionCastOp>(loc, resultType, inputs)
         .getResult(0);
+  };
+  addSourceMaterialization(addUnrealizedCast);
+  addTargetMaterialization(addUnrealizedCast);
+
+  /// String conversion
+  addConversion([](StringType stringType) {
+    return Torch::StringType::get(stringType.getContext());
   });
 
-  addTargetMaterialization([&](OpBuilder &builder, Type resultType,
-                               ValueRange inputs,
-                               Location loc) -> Optional<Value> {
-    if (inputs.size() != 1)
-      return llvm::None;
-    return builder.create<UnrealizedConversionCastOp>(loc, resultType, inputs)
-        .getResult(0);
+  /// Tensor conversion
+  addConversion([](TensorType type) -> Optional<Type> {
+    return Torch::ValueTensorType::get(
+      type.getContext(), type.getShape(), type.getElementType());
+  });
+  auto addTensorCast = [](OpBuilder &builder, TensorType type,
+                          ValueRange inputs, Location loc) -> Value {
+    assert(inputs.size() == 1);
+    assert(inputs[0].getType().isa<Torch::BaseTensorType>());
+    return builder.create<torch::TorchConversion::ToBuiltinTensorOp>(loc, inputs[0]);
+  };
+  addSourceMaterialization(addTensorCast);
+  addArgumentMaterialization(addTensorCast);
+  addTargetMaterialization([](OpBuilder &builder, Torch::ValueTensorType type,
+                              ValueRange inputs, Location loc) -> Value {
+    assert(inputs.size() == 1);
+    assert(inputs[0].getType().isa<TensorType>());
+    return builder.create<torch::TorchConversion::FromBuiltinTensorOp>(
+      loc, type, inputs[0]);
+  });
+
+  /// Create tensor to value tensor type conversion and ensure that
+  /// we always use signed integer types. i64 -> si64 conversion.
+  addConversion([](RankedTensorType type) -> Type {
+    Type elementType = type.cast<TensorType>().getElementType();
+    if (type.getElementType().isSignlessInteger(64)) {
```

Comment:
Probably, all signless integers should be converted to signed integers. More importantly, not all signed integeres are AnyTorchScalarType. So, just like with the floats, maybe we need to "snap" to 32, 64, ... bits?

For both tasks, we could use the type converter!

Reply:
True; just going to support it through `getIntOrFloatBitWidth`.